Add article "Autofill: Inspect and debug saved addresses" (#3392)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/autofill/index.md

Use the **Autofill** tool to inspect and debug address info that's been saved to Microsoft Edge.

The **Autofill** tool records various events that the browser triggers when a user is filling in a form by using the browser's autofill feature -- for example, when the browser remembers your address and autofills it in an address field.

<!-- ====================================================================== -->
## Overview

The **Autofill** tool provides a convenient way to automatically fill forms on websites with saved addresses.  The **Autofill** tool in DevTools lets you inspect the mapping between your form fields, predicted autofill values, and saved data.

The **Autofill** tool lets you inspect only the address info.

See also:
* [Autofill](https://web.dev/learn/forms/autofill) at web.dev.

<!-- ====================================================================== -->
## Save address info to Edge
<!-- https://developer.chrome.com/docs/devtools/autofill#save-to-chrome -->

By default, Microsoft Edge prompts you to save address info you entered into a web form when you submit it.

<!-- todo: find in Edge Settings -->

> <!-- ![A prompt to save address info](todo: autofill-prompt.png) -->

Author: add two pngs

> <!-- If there's no such prompt, in Edge, select Customize and control > Password and Autofill > Addresses and more, and turn on Save and fill addresses. You can also add new addresses here. -->

Author: Does Edge support Autofill?
What's the Edge equiv of **Settings and more** > **Password and Autofill**?
@captainbrosset

Rendered article for review:
* **Autofill: Inspect and debug saved addresses**
   * `/devtools-guide-chromium/autofill/index.md`
   * Internal preview: https://review.learn.microsoft.com/microsoft-edge/devtools-guide-chromium/autofill/index?branch=pr-en-us-3392
   * External preview: https://github.com/MicrosoftDocs/edge-developer/blob/user/mikehoffms/autofill/microsoft-edge/devtools-guide-chromium/autofill/index.md
   * Upstream: https://developer.chrome.com/docs/devtools/autofill

> <!-- ![The 'Addresses and more' menu option](todo: addresses-menu-option.png) -->

Author: Add png: The 'Addresses and more' menu option -- `addresses-menu-option.png`

---
title: "Autofill: Inspect and debug saved addresses"
description: Inspect and debug address info that's been saved to Microsoft Edge.
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.service: microsoft-edge
ms.subservice: devtools
ms.date: 02/28/2025
---
<!-- Copyright Sofia Emelianova

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       https://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.  -->
# Autofill: Inspect and debug saved addresses
<!-- https://developer.chrome.com/docs/devtools/autofill -->

> Use the **Autofill** tool to inspect and debug address info that's been saved to Microsoft Edge.

Author: ```suggestion
Use the **Autofill** tool to inspect and debug the information that's been saved to Microsoft Edge.
```
copied PB comment from PR 3378

The **Autofill** tool records various events that the browser triggers when a user is filling in a form by using the browser's autofill feature -- for example, when the browser remembers your address and autofills it in an address field.

![The Autofill tool](./index-images/autofill-tool.png)

<!-- ====================================================================== -->
## Overview

> The **Autofill** tool provides a convenient way to automatically fill forms on websites with saved addresses.  The **Autofill** tool in DevTools lets you inspect the mapping between your form fields, predicted autofill values, and saved data.

Author: ```suggestion
The browser autofill feature provides a convenient way to automatically fill forms on websites with saved addresses. The **Autofill** tool in DevTools lets you inspect the mapping between your form fields, predicted autofill values, and saved data.
```
Maybe link to https://support.microsoft.com/en-us/microsoft-edge/automatically-fill-info-in-microsoft-edge-81da697c-9910-d9b8-d50a-1712d96f3db8 to explain what the browser autofill feature is.

moved PB [Comment from 3378](https://github.com/MicrosoftDocs/edge-developer/pull/3378/files/88196323953a6046e0faf6f2bfd45ef5f3de0331#r1975694937) to here.

The **Autofill** tool lets you inspect only the address info.

See also:
* [Autofill](https://web.dev/learn/forms/autofill) at web.dev.

<!-- ====================================================================== -->
## Save address info to Edge
<!-- https://developer.chrome.com/docs/devtools/autofill#save-to-chrome -->

By default, Microsoft Edge prompts you to save address info you entered into a web form when you submit it.

> <!-- todo: find in Edge Settings -->

Author: PB comment in PR 3378, moving to here:
> See my previous comment [a few lines above], I found a SMC page with information about this.

<!-- ![A prompt to save address info](todo: autofill-prompt.png) -->

<!-- If there's no such prompt, in Edge, select Customize and control > Password and Autofill > Addresses and more, and turn on Save and fill addresses. You can also add new addresses here. -->

<!-- ![The 'Addresses and more' menu option](todo: addresses-menu-option.png) -->

If you don't have address data saved and don't want to save it, you can use test address data provided by the **Autofill** tool.

<!-- ====================================================================== -->

> ## Open the **Autofill** tool

Author: ```suggestion
## Open the Autofill tool
```

To manually open the **Autofill** tool:

1. Open DevTools.

1. In the **Activity Bar** at top, select **More tools** > **Autofill**.

To open the **Autofill** tool from the **Command Menu**:

1. Open the **Command Menu** by pressing:

   * macOS: **Command+Shift+P**
   * Windows, Linux: **Control+Shift+P**

   <!-- ![Command Menu with "autofill" entered](todo: command-menu-autofill.png) -->

1. Start typing **autofill**, select **Show Autofill**, and the press **Enter**.

   By default, the **Autofill** tool opens in **Quick View** at the bottom of DevTools.  You can move it to the **Address Bar** at top.

<!-- ====================================================================== -->
## Inspect autofill data
<!-- https://developer.chrome.com/docs/devtools/autofill#inspect -->

To inspect autofill data:

1. Make sure [Autofill is turned on and you have address info saved](#save-address-info-to-edge) in Microsoft Edge.  Alternatively, you can use test address data as described in the next section.

> 1. Open DevTools; for example, open [Demo page with accessibility issues](https://microsoftedge.github.io/Demos/devtools-a11y-testing/) in a new window or tab, right-click the page, and then select **Inspect**.

Author: This is not a great demo page for autofill, because it doesn't have a form which would trigger the browser's autofill feature.
I suggest considering this out of scope for this PR for now. We could create a new demo page and then rewrite this article to match with it, but we should do this in a separate PR. For now, you can probably just comment out everything after the **Open the Autofill tool** section

moved the above PB comment from PR 3378 to here
